src/io.cr

```diff
@@ -1155,8 +1155,20 @@ abstract class IO
   # io2.to_s # => "hello"
   # ```
   def self.copy(src, dst) : Int64
-    buffer = uninitialized UInt8[DEFAULT_BUFFER_SIZE]
     count = 0_i64
+    {% if LibC.has_method?(:copy_file_range) %}
+      if src.is_a?(File) && dst.is_a?(File)
```

Comment:
Googling to figure out why this is limited to `File` - the man pages don't say anything about it not working for arbitrary file descriptors, I ran into https://lwn.net/Articles/846403/ describing some of the issues with that. In particular there seems to be (or was, back in 21 when that article was written) issues with virtual file systems and reading from `/proc/*`. So I agree that `File` is the reasonable option. Perhaps there need to be some comment reflecting the motivations?

Reply:
thank you, good comments, i added so that if copy_file_range returns 0 we fallback, like go and rust does.

Reply:
https://github.com/rust-lang/rust/commit/4ddedd521418d67e845ecb43dc02c09b0af53022#diff-4a74019cd4088485a9e0706be6ccf93ff091cb19d9aad474e4e9d03e2bb8522d